c_glib/parquet-glib/arrow-file-writer.cpp

```diff
@@ -32,10 +32,274 @@ G_BEGIN_DECLS
  * @short_description: Arrow file writer class
  * @include: parquet-glib/parquet-glib.h
  *
+ * #GParquetWriterProperties is a class for the writer properties.
  * #GParquetArrowFileWriter is a class for writer Apache Arrow data to
  * file as Apache Parquet format.
  */
 
+typedef struct GParquetWriterPropertiesPrivate_ {
+  std::shared_ptr<parquet::WriterProperties> properties;
+  parquet::WriterProperties::Builder *builder;
+  gboolean changed;
+} GParquetWriterPropertiesPrivate;
+
+G_DEFINE_TYPE_WITH_PRIVATE(GParquetWriterProperties,
+                           gparquet_writer_properties,
+                           G_TYPE_OBJECT)
+
+#define GPARQUET_WRITER_PROPERTIES_GET_PRIVATE(object) \
+  static_cast<GParquetWriterPropertiesPrivate *>(      \
+    gparquet_writer_properties_get_instance_private(   \
+      GPARQUET_WRITER_PROPERTIES(object)))
+
+static void
+gparquet_writer_properties_finalize(GObject *object)
+{
+  auto priv = GPARQUET_WRITER_PROPERTIES_GET_PRIVATE(object);
+
+  priv->properties = nullptr;
+  delete priv->builder;
+
+  G_OBJECT_CLASS(gparquet_writer_properties_parent_class)->finalize(object);
+}
+
+static void
+gparquet_writer_properties_init(GParquetWriterProperties *object)
+{
+  auto priv = GPARQUET_WRITER_PROPERTIES_GET_PRIVATE(object);
+  priv->builder = new parquet::WriterProperties::Builder();
+  priv->changed = TRUE;
+}
+
+static void
+gparquet_writer_properties_class_init(GParquetWriterPropertiesClass *klass)
+{
+  auto gobject_class = G_OBJECT_CLASS(klass);
+
+  gobject_class->finalize     = gparquet_writer_properties_finalize;
+}
+
+/**
+ * gparquet_writer_properties_new:
+ *
+ * Return: A newly created #GParquetWriterProperties.
+ *
+ * Since: 1.0.0
+ */
+GParquetWriterProperties *
+gparquet_writer_properties_new(void)
+{
+  auto writer_properties = g_object_new(GPARQUET_TYPE_WRITER_PROPERTIES,
+                                        NULL);
+  return GPARQUET_WRITER_PROPERTIES(writer_properties);
+}
+
+/**
+ * gparquet_writer_properties_set_compression:
+ * @properties: A #GParquetWriterProperties.
+ * @compression_type: A #GArrowCompressionType.
+ *
+ * Since: 1.0.0
+ */
+void
+gparquet_writer_properties_set_compression(GParquetWriterProperties *properties,
+                                           GArrowCompressionType compression_type)
+{
+  auto arrow_compression_type = garrow_compression_type_to_raw(compression_type);
+  auto priv = GPARQUET_WRITER_PROPERTIES_GET_PRIVATE(properties);
+  priv->builder->compression(arrow_compression_type);
+  priv->changed = TRUE;
+}
+
+/**
+ * gparquet_writer_properties_get_compression:
```

Comment:
Could you append `_dot_string`?
Because we will add `parquet::Schema::ColumnPath` version later.